### PR TITLE
Fix #371

### LINKED ISSUE #371: swarm logs returns 404 when being called on components
Example:
```
swarm status sharelatex
Service sharelatex is up

component   image                         instanceid    created              status
mongo       mongo:2.4                     j5hhpe4y8cnl  2015-08-17 12:04:19  up
redis       redis:latest                  p3s63kxjcky9  2015-08-17 12:04:26  up
sharelatex  sharelatex/sharelatex:latest  b9k1cw7eqlg1  2015-08-17 12:04:34  up
```
then
```
swarm logs sharelatex/sharelatex
404 page not found
```
and
```
swarm logs sharelatex/redis
404 page not found
```
but
```
swarm logs hello/me
Please provide an instance ID. Use --help for details.
```
and
```
swarm logs sharelatex
Please provide an instance ID. Use --help for details.
```

### PR DESCRIPTION
**Pull request by marians**
Fixes #371

This handles the case where the given argument, either with or without slash, cannot be resolved to an instance ID.

Preview:

```
The argument 'wesdfsdffddffds' can't be resolved to an instance ID.
Please provide a valid instance ID or the first part of one.
```

```
The argument 'wesdfs/sdsdf' can't be resolved to an instance ID.
Please provide a valid instance ID or the first part of one.
```

RFR @zyndiecate @ewoutp

### CODE REVIEW

**Comment by zyndiecate** [comment]
I personally don't like too much text, but you are the UX guy. Just saying.

File: ids/partial_ids.go [hunk 1]
```diff
@@ -43,8 +43,14 @@ func ParseIdArg(arg string, options *IdArgumentOptions, client *apiclient.Client
...
 	parts := strings.Split(arg, "/")
 
-	// Check for multiple parts. If o it can only be a component.
+	// Check for multiple parts. If so it can only be a component.
 	if len(parts) > 1 {
+		if !options.AllowComponent {
+			if options.AllowInstanceId {
+				return nil, maskAny(InvalidInstanceIdArgumentError)
+			}
+			// TODO: handle other specific errors
```
Comment (zyndiecate) [inline ids/partial_ids.go]: What kind of errors? When to do so?

**Comment by zyndiecate** [comment]
Minor questions. LGTM

File: ids/partial_ids.go (hunk 1 above)
Comment (marians) [inline ids/partial_ids.go]: Actually I meant the case where `options.AllowInstanceId` might be false but `options.AllowService` would be true. But this is currently not a valid assumption, so I'll change this bit.

File: logs.go [hunk 2]
```diff
@@ -12,9 +12,12 @@ import (
 )
 
 const (
-	logsTailAll        = "all"
-	defaultLogLines    = 10
-	logsInvalidArgsMsg = "Please provide an instance ID. Use --help for details.\n"
+	logsTailAll                     = "all"
+	defaultLogLines                 = 10
+	logsInvalidArgsMsg              = "Please provide an instance ID. Use --help for details.\n"
+	invalidInstanceIdFormatTemplate = `The argument '%s' can't be resolved to an instance ID.
```
Comment (ewoutp) [inline logs.go]: I don't like that this constant is in logs.go while it is a generic message.
We should move it to a file with generic messages (which we do not have yet)

Comment (ewoutp) [inline logs.go]: See #468

**Pull request merged**